packages/lib/src/index.ts

```diff
@@ -34,7 +13,43 @@ export type HTMLElementProps<
   ? Props
   : never;
 
-export { typeIs, testHelpers };
+/**
+ * Helper that constructs a type requiring at least one of the passed keys
+ * to be present in the passed interface.
+ *
+ * Example
+ * ```
+ * interface ExampleInterface {
+ *   alwaysRequired: boolean,
+ *   sometimesRequired: boolean,
+ *   requiredOtherTimes: boolean,
+ * }
+ *
+ * type ExampleEither = Either<ExampleInterface, 'sometimesRequired' | 'requiredOtherTimes'>
+ *
+ * // The above is equivalent to:
+ * interface SharedInExampleInterface {
+ *   alwaysRequired: boolean,
+ * }
+ *
+ * interface FirstIsRequired extends SharedInExampleInterface {
+ *   sometimesRequired: boolean,
+ *   requiredOtherTimes?: boolean,
+ * }
+ *
+ * interface SecondIsRequired extends SharedInExampleInterface {
+ *   sometimesRequired?: boolean,
+ *   requiredOtherTimes: boolean,
+ * }
+ *
+ * type EquivalentToExampleEither = FirstIsRequired | SecondIsRequired
+ * ```
+ */
+export type Either<T, Keys extends keyof T = keyof T> = Omit<T, Keys> &
+  {
+    [K in Keys]-?: Required<Pick<T, K>> &
+      Partial<Record<Exclude<Keys, K>, undefined>>;
```

Comment:
Shouldn't this be `Partial<Pick<T, Exclude<Keys, K>>>`? At least, I'm pretty sure that would match your example. As it is, if I'm reading it right, this would be equivalent to

```ts
interface FirstIsRequired extends SharedInExampleInterface {
  sometimesRequired: boolean;
  requiredOtherTimes?: undefined;
}
```

with `undefined` instead of `boolean`. Of course, if the goal is to disallow people from providing _both_ keys, then this is right and the example should be updated.

Reply:
Hmm... I think you're right here – good catch! I tested, and confirmed this change works. As is, it was requiring that only one of the two was passed when really having both is okay.

Reply:
Just pushed the fix 😁